pkg/pipeline/encode/encode_prom.go

```diff
@@ -102,9 +104,15 @@ func (e *encodeProm) EncodeMetric(metric config.GenericMap) []config.GenericMap
 	// TODO: We may need different handling for histograms
 	out := make([]config.GenericMap, 0)
 	for metricName, mInfo := range e.metrics {
+		val, keyFound := metric[mInfo.filterKey]
+		shouldKeepRecord := keyFound && val == mInfo.filterValue
```

Comment:
A very fast "thing" that you can improve here is to say ... if there is a Key and the value equals to some N/A value we filter just on the basis of the existence of the Key and we do not care about the value of the value ... trivial change and adds a lot of flexibility

Reply:
I see what you are suggesting and I agree that it will add flexibility. But, I think we should put more thought into that.
I don't like magic values that might collide with the user's values.